woocommerce/google-listings-and-ads
# Setup MC Step 1 - Overwrite existing product feed UI

### Changes proposed in this Pull Request:

Frontend part of #280 . Note that this PR targets the `feature/existing-product-feed-notice` branch and not the `trunk` branch; after merging, this would be part of https://github.com/woocommerce/google-listings-and-ads/pull/345.

This PR provides the UI for overwriting product feed when connecting Google MC account in Setup MC Step 1. The `OverwriteFeedCard` would be displayed when there is a HTTP 409 with `action: 'feed-overwrite'` in the body in response to the following requests: 

- `POST /wc/gla/mc/accounts` with body `{ id }` (i.e. linking an existing MC account)
- `POST /wc/gla/mc/accounts/switch-url` with body `{ id }` (i.e. switch URL request)

### Screenshots:

![image](https://user-images.githubusercontent.com/417342/111978757-3e56ab80-8b3f-11eb-8f23-7e96200cc750.png)

### Detailed test instructions:

1. In an MC account with a claimed website URL, create a new primary product feed (not "Content API" input method): https://merchants.google.com/mc/products/sources/createPrimaryFeed
2. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc
3. Open your browser dev tools network panel.
4. In the Step 1 UI, Connect Jetpack, Google Account and Google MC account. The Google MC account should be the above account with an existing product feed.
5. When you connect to the Google MC account, in the network panel, confirm that there is a HTTP 409 response with the message indicating existing feeds, and "action":"feed-overwrite".
6. Click on the "Overwrite my feed" button.
7. You should see a card with your Google MC account number indicating setup is complete, and you should be able to proceed to the next step by clicking on the continue button.

Bonus: Same, but with an MC account with a different claimed website URL (will require clicking "Switch to my new URL" button first, and then "Overwrite my feed" button).
